config/filter.d/common.conf

```diff
@@ -44,14 +44,16 @@ __md5hex = (?:[\da-f]{2}:){15}[\da-f]{2}
 
 # bsdverbose is where syslogd is started with -v or -vv and results in <4.3> or
 # <auth.info> appearing before the host as per testcases/files/logs/bsd/*.
-__bsd_syslog_verbose = (<[^.]+\.[^.]+>)
+__bsd_syslog_verbose = <[^.]+\.[^.]+>
+
+__date_ambit = \[\]
 
 # Common line prefixes (beginnings) which could be used in filters
 #
 #      [bsdverbose]? [hostname] [vserver tag] daemon_id spaces
 #
 # This can be optional (for instance if we match named native log files)
-__prefix_line = \s*%(__bsd_syslog_verbose)s?\s*(?:%(__hostname)s )?(?:%(__kernel_prefix)s )?(?:@vserver_\S+ )?%(__daemon_combs_re)s?\s%(__daemon_extra_re)s?\s*
+__prefix_line = (?:%(__date_ambit)s)?\s*(?:(?:%(__bsd_syslog_verbose)s)\s*)?(?:(?:%(__hostname)s)\s*)?(?:(?:%(__kernel_prefix)s)\s*)?(?:(?:@vserver_\S+)\s*)?(?:(?:%(__daemon_combs_re)s)\s*)?(?:(?:%(__daemon_extra_re)s)\s*)?
```

Comment:
So far your example only confirms what I have recommended -- to use groups within variable definitions. E.g. in your case if you had defined `a=(?:b|c)` you would be just fine in your `x` definition, wouldn't you?
It is because this way all possible logical operands applied within variable definition have a clear defined boundary, and an evil someone can then use it without realizing that he needed to surround it by an additional group just to guarantee correct application of logical operrands within it!

But indeed, if we are  to please both of our paranoias we would ultimately need both -- groups while defining variables and then, to prevent silly user's mistake of not using groups within his varaibles, while using those variables along side with additional characters (i.e. space in this case ;-) ).  But it would render many regexps hard to comprehend.  In our particular case here though we rely on variable definitions which aren't generally modified or passed from jail specs somehow but simply used for string substitutions pretty much right here.  Thus IMHO we could safely stick to previously used groupping within variables definitions, which will assure that any logical operand within them, would have it bound just fine and trailng space added will be applied outside of that boundary without  needing double-protection.
Makes sense, or am I still missing? ;)